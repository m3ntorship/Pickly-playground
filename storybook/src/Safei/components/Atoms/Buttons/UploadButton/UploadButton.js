import React from "react";
const UploadButton = () => {
  return (
    <button className="border border-dashed border-accent bg-grey-shd7 rounded-md text-accent focus:outline-none flex justify-center items-center mx-2 sm:mx-0 max-w-xl w-full h-14">
      <svg
        className="mr-1 w-4 h-4 fill-current"
        viewBox="0 0 24 24"
        fill="currentColor"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 5H17V12H19V5C19 3.897 18.103 3 17 3H4C2.897 3 2 3.897 2 5V17C2 18.103 2.897 19 4 19H12V17H4V5Z"
          fill="currentColor"
        />
        <path d="M8 11L5 15H16L12 9L9 13L8 11Z" fill="currentColor" />
        <path
          d="M19 14H17V17H14V19H17V22H19V19H22V17H19V14Z"
          fill="currentColor"
        />
      </svg>
      <span className="">Upload one or multiple images</span>
    </button>
  );
};

export default UploadButton;
