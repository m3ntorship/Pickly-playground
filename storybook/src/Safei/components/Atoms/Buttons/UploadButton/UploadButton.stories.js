import React from "react";
import "@storybook/addon-actions";

import UploadButton from "./UploadButton";

export default {
  title: "Atoms/Buttons/UploadButton",
  component: UploadButton,
  // parameters: { actions: { argTypesRegex: "^on.*" } },
  // argTypes: { onClick: { action: "clicked" } },
};

const Template = (args) => <UploadButton {...args} />;
export const Upload = Template.bind({});
Upload.args = {
  //   disabled: false,
  //   size: "lg",
};
