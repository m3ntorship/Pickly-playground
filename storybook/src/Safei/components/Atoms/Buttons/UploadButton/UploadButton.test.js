import React from "react";
import renderer from "react-test-renderer";
import UploadButton from "./UploadButton";

describe("UploadButton  field", () => {
  it("renders  UploadButton ", () => {
    const tree = renderer.create(<UploadButton status="error" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
