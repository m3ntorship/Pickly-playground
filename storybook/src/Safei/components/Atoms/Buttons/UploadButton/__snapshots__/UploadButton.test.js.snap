// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadButton  field renders  UploadButton  1`] = `
<div
  className="border-2 border-dashed border-accent bg-grey-shd7 rounded-md text-accent focus:outline-none relative h-14 text-center"
  id="upload-container"
>
  <span
    className="text-accent object-none absolute top-3.5 left-1/4 px-7 font-medium font-sm"
  >
    <svg
      className="mr-1 w-4 h-4 fill-current inline"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 5H17V12H19V5C19 3.897 18.103 3 17 3H4C2.897 3 2 3.897 2 5V17C2 18.103 2.897 19 4 19H12V17H4V5Z"
        fill="currentColor"
      />
      <path
        d="M8 11L5 15H16L12 9L9 13L8 11Z"
        fill="currentColor"
      />
      <path
        d="M19 14H17V17H14V19H17V22H19V19H22V17H19V14Z"
        fill="currentColor"
      />
    </svg>
    Upload one or multiple images
  </span>
  <input
    className="opacity-0 min-w-full h-full cursor-pointer"
    data-testId="upload-btn"
    id="myFile"
    multiple={true}
    name="filename"
    type="file"
  />
</div>
`;
